flocker/node/agents/testtools/_cinder.py

```diff
@@ -58,3 +64,46 @@ def setUp(self):
             self.client = client_factory(test_case=self)
 
     return Tests
+
+
+def mimic_for_test(test_case):
+    """
+    Start a mimic server in the background on an ephemeral port and return the
+    port number.
+
+    This is used in synchronous test cases so I can't launch the mimic service
+    in process.
+
+    Parsing the logs for the chosen port number is ugly, but ``find_free_port``
+    kept returning ports that were in use when mimic attempted to bind to them.
+    """
+    log = test_case.make_temporary_path()
+    stdout = test_case.make_temporary_path()
+    stderr = test_case.make_temporary_path()
+    with stdout.open('w') as stdout, stderr.open('w') as stderr:
+        p = Popen(['twistd', '--nodaemon', '--logfile', log.path,
+                   'mimic', '--listen', '0', '--realtime'],
```

Comment:
It tells mimic to listen on *any* free port. 
Then I have to parse the logs to discover what port has been used....which is ugly, but when I tried to supply a port number chosen by ``findFreePort`` I found that mimic would frequently fail to bind to that port....perhaps because ``findFreePort`` hadn't freed the port in time.
@exarkun suggested using systemd style inherited listening sockets, but I haven't had time to try that out yet.